lib/internal/test_runner/tap_stream.js

```diff
@@ -46,26 +48,38 @@ class TapStream extends Readable {
     }
   }
 
-  bail(message) {
-    this.#tryPush(`Bail out!${message ? ` ${tapEscape(message)}` : ''}\n`);
-  }
-
-  fail(indent, testNumber, name, duration, error, directive) {
-    this.emit('test:fail', { __proto__: null, name, testNumber, duration, ...directive, error });
+  fail(indent, testNumber, name, details, directive) {
+    this.emit('test:fail', {
```

Comment:
need to change the corresponding documentation for the `test:fail` and `test:pass` events

Reply:
done.